src/annotator/frame-observer.js

```diff
@@ -124,8 +124,9 @@ export function onDocumentReady(frame) {
       frame.hasAttribute('src') &&
       frame.src !== 'about:blank'
     ) {
-      // Unfortunately, listening for 'DOMContentLoaded' on the iframeDocument
-      // doesn't work. Instead, we need to wait for a 'load' event to be triggered.
+      // Listening to `document.DOMContentLoaded` doesn't work because the
+      // document is replaced. On the other hand, listening to `window.load`
+      // works because window doesn't change.
```

Comment:
I could add a reference to https://github.com/hypothesis/client/pull/3748 about the failed efforts to avoid `window.load`